File: app/src/main/java/workshop1024/com/xproject/controller/activity/SettingsActivity.java
package workshop1024.com.xproject.controller.activity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import workshop1024.com.xproject.R;

public class SettingsActivity extends AppCompatActivity {
    private Toolbar mToolbar;

    public static void startActivity(Context context) {
        Intent intent = new Intent(context, SettingsActivity.class);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.settings_activity);

        mToolbar = findViewById(R.id.settings_toolbar_navigator);
        setSupportActionBar(mToolbar);

        ActionBar actionBar = getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);
    }
}

File: settings/src/main/java/workshop1024/com/xproject/settings/controller/activity/SettingsActivity.kt
package workshop1024.com.xproject.settings.controller.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import workshop1024.com.xproject.base.utils.UnitUtils
import workshop1024.com.xproject.settings.R
import workshop1024.com.xproject.settings.controller.fragment.SettingsFragment

class SettingsActivity : AppCompatPreferenceActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        val headerListView = listView
        headerListView.divider = getDrawable(R.drawable.setting_headerlist_itemdivider)
        headerListView.dividerHeight = UnitUtils.spToPx(this, 1f)
    }

    override fun onBuildHeaders(target: List<Header>) {
        loadHeadersFromResource(R.xml.settings_preferences, target)
    }

    override fun isValidFragment(fragmentName: String): Boolean {
        return SettingsFragment::class.java.name == fragmentName
    }

    companion object {

        fun startActivity(context: Context) {
            val intent = Intent(context, SettingsActivity::class.java)
            context.startActivity(intent)
        }
    }
}
